Editors/RoomEditor.cpp

```diff
@@ -1,13 +1,141 @@
+  resMapper->addMapping(ui->clearCheckBox, Room::kClearDisplayBufferFieldNumber);
+  resMapper->addMapping(ui->persistentCheckBox, Room::kPersistentFieldNumber);
+  resMapper->addMapping(ui->captionLineEdit, Room::kCaptionFieldNumber);
+
+  resMapper->addMapping(ui->enableViewsCheckBox, Room::kEnableViewsFieldNumber);
+  resMapper->addMapping(ui->clearViewportCheckBox, Room::kClearViewBackgroundFieldNumber);
+  resMapper->toFirst();
+
+  RepeatedProtoModel* vm = roomModel->GetRepeatedSubModel(Room::kViewsFieldNumber);
+  ImmediateDataWidgetMapper* viewMapper = new ImmediateDataWidgetMapper(this);
+  viewMapper->setModel(vm);
+  viewMapper->addMapping(ui->viewVisibleCheckBox, View::kVisibleFieldNumber);
+
+  viewMapper->addMapping(ui->cameraXSpinBox, View::kXviewFieldNumber);
+  viewMapper->addMapping(ui->cameraYSpinBox, View::kYviewFieldNumber);
+  viewMapper->addMapping(ui->cameraWidthSpinBox, View::kWviewFieldNumber);
+  viewMapper->addMapping(ui->cameraHeightSpinBox, View::kHviewFieldNumber);
+
+  viewMapper->addMapping(ui->viewportXSpinBox, View::kXportFieldNumber);
+  viewMapper->addMapping(ui->viewportYSpinBox, View::kYportFieldNumber);
+  viewMapper->addMapping(ui->viewportWidthSpinBox, View::kWportFieldNumber);
+  viewMapper->addMapping(ui->viewportHeightSpinBox, View::kHportFieldNumber);
+
+  viewMapper->addMapping(ui->followingHBorderSpinBox, View::kHborderFieldNumber);
+  viewMapper->addMapping(ui->followingVBorderSpinBox, View::kVborderFieldNumber);
+  viewMapper->addMapping(ui->followingHSpeedSpinBox, View::kHspeedFieldNumber);
+  viewMapper->addMapping(ui->followingVSpeedSpinBox, View::kVspeedFieldNumber);
+  viewMapper->toFirst();
+
+  QMenuView* objMenu = new QMenuView(this);
+  TreeSortFilterProxyModel* treeProxy = new TreeSortFilterProxyModel(this);
+  treeProxy->SetFilterType(TreeNode::TypeCase::kObject);
+  treeProxy->setSourceModel(MainWindow::treeModel);
+  objMenu->setModel(treeProxy);
+  ui->objectSelectButton->setMenu(objMenu);
+
+  connect(objMenu, &QMenu::triggered, this, &RoomEditor::SelectedObjectChanged);
+
+  connect(ui->currentViewComboBox, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged),
+          [=](int index) { viewMapper->setCurrentIndex(index); });
+
+  RepeatedProtoModel* m = roomModel->GetRepeatedSubModel(Room::kInstancesFieldNumber);
+  ui->layersAssetsView->setModel(m);
+  connect(ui->layersAssetsView->selectionModel(), &QItemSelectionModel::selectionChanged,
+          [=](const QItemSelection& selected, const QItemSelection& /*deselected*/) {
+            if (selected.empty()) return;
+            auto selectedIndex = selected.indexes().first();
+            auto currentInstanceModel =
+                roomModel->data(Room::kInstancesFieldNumber, selectedIndex.row()).value<void*>();
+            ui->layersPropertiesView->setModel(static_cast<ProtoModel*>(currentInstanceModel));
+          });
+
+  for (int c = 0; c < m->columnCount(); ++c) {

```

Comment:
This also looks general... but I can't tell what it's for.

Reply:
This is just fundies trying to hide certain columns. Because the properties view shows the more detailed instance properties, he doesn't want the instance list showing them all either. He just wants the instance list to show name/id/sprite. I admit, it could be generalized, and he may have actually already generalized most of it.